.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,289 @@
+name: test-and-build 
+on:
+  push:
+
+env:
```

Comment:
There is a limitation in Github Actions where you cannot pass an environment variable to a reusable workflows. You will the GOTESTSUM_VERSION being duplicated in several files because I cannot pass ${{ env.GOTESTSUM_VERSION }} into the inputs of reusable workflows. It is the same with go-version. I would have rather set a variable for that but it isn't possible 👎